src/main/java/codeu/controller/TestDataServlet.java

```diff
@@ -75,7 +75,9 @@ void setUserStore(UserStore userStore) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-    request.getRequestDispatcher("/WEB-INF/view/testdata.jsp").forward(request, response);
+    if(request.getSession().getAttribute("user").isSuperUser()){
```

Comment:
This is actually set in LoginServlet as a String attribute, so this will error out. See https://github.com/hsha0/CodeU-Spring2018-Team19/blob/master/src/main/java/codeu/controller/LoginServlet.java#L75. Also see https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#setAttribute-java.lang.String-java.lang.Object- for reference too.

You need to change "setAttribute" there to store the actual User object. Also, here and in other places you need to typecast to User. Like:
User currentUser = (User) request.getSession().getAttribute("user");
if (currentUser.isSuperUser()) .....